server/remote_cache/byte_stream_server/byte_stream_server.go

```diff
@@ -108,6 +109,20 @@ func (s *ByteStreamServer) ReadCASResource(ctx context.Context, r *digest.CASRes
 		return err
 	}
 
+	readDone := make(chan struct{})
```

Comment:
Instead of adding a log on the entire Read, I think it would be more useful to add a timeout for the individual `stream.Send` operation. There you could also set a lower timeout value (like 30s) since it really shouldn't take long to send a small chunk.

Reply:
I also remembered that we have a `canary` package that exists for this purpose.

Reply:
Ack updated